bindings/matlab/+iDynTreeWrappers/prepareVisualization.m

```diff
@@ -63,7 +69,15 @@
 [linkMeshInfo,map]=iDynTreeWrappers.getMeshes(model,meshFilePrefix);
 numberOfLinks=length(linkMeshInfo);
 linkNames=cell(numberOfLinks,1);
-mainHandler=figure;
+figHandles = findobj('Type', 'figure', 'Name', options.name);
+if options.reuseFigure && size(figHandles, 1) > 0
+    mainHandler=figHandles(1,1);
+    clf(mainHandler,'reset')
+else
+    mainHandler=figure;
```

Comment:
In the case, there is no previous figure I believe is better to use gcf instead of figure. The reason is it will create a figure if none has been created. It is more compliant with other Matlab functionalities.  It also allows to manipulate all the figure properties from outside the visualizer that the visualizer will directly use. This way you can also set the name the number or any property you want and the visualizer won't care.

Reply:
Sorry for merging so quick @fjandrad ! @S-Dafarra if @fjandrad comment make sense feel free to submit another PR, no need to update the CHANGELOG in that case.

Reply:
> In the case, there is no previous figure I believe is better to use `gcf` instead of `figure`. The reason is it will create a figure if none has been created. It is more compliant with other Matlab functionalities.  It also allows to manipulate all the figure properties from outside the visualizer that the visualizer will directly use. This way you can also set the name the number or any property you want and the visualizer won't care. 

But in this case, if there is a figure already opened that I don't want to touch, I would start modifying that figure right?

Reply:
Yes, it's true which happens with all other plotting functions of Matlab. You can keep the logic for when you send a name. That way you keep both functionalities.

Reply:
Fully agree @fjandrad . Anyway, if a figure is already open, and which we want to leave intact, we just need to create a new one with the desired properties before calling the `prepareVisualization` function.

Reply:
@S-Dafarra , last but not least, I would add a parameter`figureNum` for reusing the figure wih that number (the field in the **Figure** class is `Number`). This allows to reuse the same figure frame that we configured and sized outside the visualizer, through multiple runs, while changing the name at each run. This can be useful, for instance, if the name of the figure reflects some critical parameters of the test.

I suggest this since you are going to submit a new pull request, and the required changes for this are really minimal.

Reply:
> Yes, it's true which happens with all other plotting functions of Matlab. You can keep the logic for when you send a name. That way you keep both functionalities
>  Fully agree @fjandrad . Anyway, if a figure is already open, and which we want to leave intact, we just need to create a new one with the desired properties before calling the prepareVisualization function.

To be honest, I don't really like this. It is not obvious that the visualizer will need a figure, and if the behavior changes according to one option non trivially, this may create a lot of confusion. Suppose that you enable the "reuse" figure from a struct. This would require the user to deal with the different cases according to the value of the struct. 
Moreover, this behavior would break the compatibility for people already using this.

> @S-Dafarra , last but not least, I would add a parameterfigureNum for reusing the figure wih that number (the field in the Figure class is Number). This allows to reuse the same figure frame that we configured and sized outside the visualizer, through multiple runs, while changing the name at each run. This can be useful, for instance, if the name of the figure reflects some critical parameters of the test.

I don't like this either 😬 Initially, I was looking for some hidden identifier, but then I realized that setting the name is indeed pretty intuitive for the user and easy to document: "same name, same figure". For the number instead, imho is more difficult and less intuitive to find unique numbers. Plus, Matlab increments the number at every new figure, so there is a higher chance of using a figure already opened. in addition, what would be a "default" number. Right now, we can enable this feature even without setting a name.

I know that fixing the name may conflict with some scripts, but at the end, if we need to change the name, the visualizer would simply work as before. After all, "different name, different figure" 😁 Luckily, Matlab allows us to set also the "Title"of the plot. So, in principle, we can change the title, while keeping the same window name.

Reply:
## gcf or no gcf ...
After our F2F discussion, we agreed that a safe option, avoiding to break existing scripts, would be to use a currently "focused" figure (in the sense of Matlab, i.e. `gcf`) as an alternative option to using an existing figure with same name or creating a new figure.

## The motivation ...
The motivation behind this approach, as mention earlier, is to be able to have the same behavior as all the plotting Matlab functions:
- either plot into the currently "focused" figure returned by `gcf`.
- either create a new figure and plot into that figure.
With respect to that, we get the additional option:
- plot into a figure with the same name passed as input parameter.

## How to do it ...
We also discussed on how to do it:
- keep the already added input parameters `name` and `reuseFigure`.
- `reuseFigure` would be a multivalue flag set following the logic below:
    - 0: no reuse. Create a figure.
    - 1: reuse figure with same name `name`.
    - 2: reuse figure returned by `gcf` (actually `gcf` creates a figure in case all have been closed).

> his would require the user to deal with the different cases according to the value of the struct.
Actually, in the third case it is the responsability of the user to "focus"on a proper figure that exists.

Reply:
Thanks @nunoguedelha for detailing this. I will open a PR for this additional option.

Reply:
Small suggestion: I would add the PR link in this conversation, before hitting "Resolve conversation".

Reply:
Here you go #705